Integrations/PerceptionPoint/PerceptionPoint.py

```diff
@@ -0,0 +1,177 @@
+        if api_res.get('next'):
+            api_res = get_pp_api_result(url=api_res.get('next'))
+            num_of_results = api_res.get('count')
+        api_loops += 1
+    return incidents
+
+
+def report_incidents(incidents_list):
+    demisto.incidents(incidents_list)
+
+
+def get_pp_api_result(url, **kwargs):
+    try:
+        res = requests.get(url=url,
+                           params=kwargs,
+                           headers=HEADER,
+                           verify=SECURED)
+        res.raise_for_status()
+        try:
+            res_content = res.json()
+        except Exception:
+            res_content = {}
+        return res_content
+    except requests.exceptions.HTTPError as err:
+        if 400 <= res.status_code < 500:
+            return_error('Invalid token')
+        else:
+            return_error(err)
+    except Exception as err:
+        return_error(err)
+
+
+def build_request_url(api_action):
+    return URL.format(endpoint=API_ACTIONS_DICT.get(api_action))
+
+
+def command_fetch_incidents():
+    try:
+        args_dict = demisto.args()
+        user_params_copy = copy.deepcopy(USER_PARAMS)
+        user_params_copy.pop('insecure')
+        user_params_copy.pop('pp_token')
+        user_params_copy.pop('api_loops')
+        args_dict.update(user_params_copy)
+        req_args = build_fetch_incident_types(args_dict)
+        last_run_id = int(demisto.getLastRun().get('scan_id', 0))
+        req_args[API_CURSOR_ARG] = last_run_id
+        incidents_list = collect_incidents(**req_args)
+        report_incidents(incidents_list)
+        if incidents_list:
+            last_run_id = max(last_run_id, int(incidents_list[-1].get('Scan Id')))
+            demisto.setLastRun({'scan_id': int(last_run_id)})
+    except Exception:
+        return_error(
+            'An error occurred while trying to fetch new incidents. Please contact PerceptionPoint support')
+
+
+def release_email_and_get_message(scan_id_to_release):
+    try:
+        release_url = build_request_url(RELEASE).format(id_=scan_id_to_release)
+        _ = get_pp_api_result(url=release_url)
+        return 'Email with id {} was released Successfully!'.format(scan_id_to_release)
+    except Exception:
+        raise
+
+
+def command_release_email():
+    try:
+        scan_id_to_release = demisto.args().get('scan_id')
+        entry = {
+            'Type': entryTypes['note'],
+            'ReadableContentsFormat': formats['markdown']
+        }
+        email_release_response = release_email_and_get_message(scan_id_to_release)
+        entry.update({'Contents': email_release_response,
+                      'ContentsFormat': formats['text'],
+                      'EntryContext': {'PP.Released': scan_id_to_release}}
+                     )
+        demisto.results(entry)
+    except Exception:
+        return_error(
+            'An error occurred while trying to release email. Please contact PerceptionPoint support')
```

Comment:
please add the str of the exception itself so you will have more info about what happened.